### PR TITLE
ios_vlan DI module

### PR DESCRIPTION
**Pull request by trishnaguha**
Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

##### SUMMARY

ios_vlan DI module https://github.com/ansible/ansible/issues/31295
Integration test

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

modules/network/ios/ios_vlan
##### ANSIBLE VERSION

```
2.5.0
```

### CODE REVIEW

**Comment by itdependsnetworks** [comment]
@trishnaguha I thought DI was in no change mode for 2.5? 

item 3
https://meetbot.fedoraproject.org/ansible-network/2017-10-18/network_working_group.2017-10-18-16.00.html

**Comment by trishnaguha** [comment]
@itdependsnetworks The DI modules we are adding in 2.5 are the modules which were supposed to be added in 2.4. So these are the missing platform specific `net_*`(platform agnostic) modules which includes DI.

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+        name=dict(),
+        interfaces=dict(type='list'),
+        delay=dict(default=10, type='int'),
+        state=dict(default='present',
+                   choices=['present', 'absent', 'active', 'suspend'])
+    )
+
+    aggregate_spec = deepcopy(element_spec)
+    aggregate_spec['vlan_id'] = dict(required=True)
+
```
Comment (ganeshrn) [inline lib/ansible/modules/network/ios/ios_vlan.py:255]: ~~imo a mutually exclusive check is required for `vlan_id` and `aggregate` params~~
Please ignore this comment

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+    required_one_of = [['vlan_id', 'aggregate']]
+    mutually_exclusive = [['vlan_id', 'aggregate']]
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           required_one_of=required_one_of,
+                           mutually_exclusive=mutually_exclusive,
+                           supports_check_mode=True)
+    warnings = list()
+    check_args(module, warnings)
```
Comment (ganeshrn) [inline lib/ansible/modules/network/ios/ios_vlan.py]: `check_args()` is no longer required.

File: test/integration/targets/ios_vlan/tests/cli/basic.yaml [hunk 3]
```diff
@@ -0,0 +1,152 @@
+#---
```
Comment (ganeshrn) [inline test/integration/targets/ios_vlan/tests/cli/basic.yaml:1]: What's the reason to comment out all the test cases in this file?

File: test/integration/targets/ios_vlan/tests/cli/basic.yaml [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+#      - "'no vlan 100' in result.commands"
+
+#- name: delete vlan(idempotence)
+#  ios_vlan: *delete
+#  register: result
+
+#- assert:
+#    that:
+#      - "result.changed == false"
+
```
Comment (ganeshrn) [inline test/integration/targets/ios_vlan/tests/cli/basic.yaml:138]: Can add a test case for aggregate parameter

**Review by ganeshrn — CHANGES_REQUESTED** [review]
Update my review comments on PR.

File: test/integration/targets/ios_vlan/tests/cli/basic.yaml (hunk 3 above)
Comment (trishnaguha) [reply test/integration/targets/ios_vlan/tests/cli/basic.yaml:1]: We do not have ios l2 switch image(15.2) to run on DCI but we have in lab. This is why we have to disable the ios_vlan test suite for now.

File: test/integration/targets/ios_vlan/tests/cli/basic.yaml (hunk 4 above)
Comment (trishnaguha) [reply test/integration/targets/ios_vlan/tests/cli/basic.yaml:138]: Added the test in new commit.

**Comment by trishnaguha** [comment]
Closing and reopening the PR to trigger the CI since it failed for some other reason.

**Review by ganeshrn — APPROVED** [review]
Looks Good

**Pull request merged**